app/controllers/posts_controller.rb

```diff
@@ -21,9 +21,11 @@ def create
     @post.users_id = current_user.id
 
     if @post.save
+      flash[:success] = 'Post saved successfully'
       redirect_to user_path(current_user.id)
     else
       render :new
+      flash.now[:error] = 'Post not saved'
```

Comment:
- [ ] In the post's index method it should solve `N+1 problem` when fetching all posts and their comments for a user.

Hint:
`@posts = @user.posts.includes(:comments)` instead of `@posts = @user.posts`